File: org.eclipse.winery.ui/src/app/instance/visualAppearance/relationshipTypesVisualsApiData.ts
/**
 * Copyright (c) 2017 University of Stuttgart.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * and the Apache License 2.0 which both accompany this distribution,
 * and are available at http://www.eclipse.org/legal/epl-v10.html
 * and http://www.apache.org/licenses/LICENSE-2.0
 *
 * Contributors:
 *     Lukas Balzer - initial API and implementation
 */

export class RelationshipTypesVisualsApiData {
    sourcearrowhead: string;
    targetarrowhead: string;
    dash: string;
    color: string;
    hovercolor: string;
    boolData: any;

    public constructor(data: any, createBools: boolean) {
        this.sourcearrowhead = data.sourcearrowhead;
        this.targetarrowhead = data.targetarrowhead;
        this.dash = data.dash;
        this.color = data.color;
        this.hovercolor = data.hovercolor;
        if (createBools) {
            this.boolData = new DataInfo();
        }
    }
}

export class DataInfo {
    dashSelected: boolean;
    targetarrowheadSelected: boolean;
    sourcearrowheadSelected: boolean;

    public constructor() {
        this.sourcearrowheadSelected = false;
        this.targetarrowheadSelected = false;
        this.dashSelected = false;
    }
}

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/resources/apiData/RelationshipTypesVisualsApiData.java
/*******************************************************************************
 * Copyright (c) 2017 University of Stuttgart.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * and the Apache License 2.0 which both accompany this distribution,
 * and are available at http://www.eclipse.org/legal/epl-v10.html
 * and http://www.apache.org/licenses/LICENSE-2.0
 *
 * Contributors:
 *     Nicole Keppler - initial API and implementation
 *******************************************************************************/
package org.eclipse.winery.repository.resources.apiData;

import org.eclipse.winery.repository.resources.entitytypes.relationshiptypes.VisualAppearanceResource;

public class RelationshipTypesVisualsApiData {
	public String sourcearrowhead;
	public String targetarrowhead;
	public String dash;
	public String color;
	public String hovercolor;

	public RelationshipTypesVisualsApiData (VisualAppearanceResource visuals) {
		this.sourcearrowhead = visuals.getSourceArrowHead();
		this.targetarrowhead = visuals.getTargetArrowHead();
		this.dash = visuals.getDash();
		this.color = visuals.getColor();
		this.hovercolor = visuals.getHoverColor();
	}

	public RelationshipTypesVisualsApiData() {

	}

	@Override
	public String toString() {
		return "{sourcearrowhead: " + sourcearrowhead + "}";
	}
}
